pkg/tsdb/prometheus/prometheus.go

```diff
@@ -163,9 +158,25 @@ func parseQuery(queries []*tsdb.Query, queryContext *tsdb.TsdbQuery) (*Prometheu
 		return nil, err
 	}
 
+	intervalFactor := queryModel.Model.Get("intervalFactor").MustInt64(1)
+
+	var step time.Duration
+	stepSeconds, err := queryModel.Model.Get("step").Int64()
+	if err != nil {
+		if err.Error() == "invalid value type" {
+			calculator := tsdb.NewIntervalCalculator(&tsdb.IntervalOptions{MinInterval: time.Second * 1})
+			interval := calculator.Calculate(queryContext.TimeRange)
+			step = interval.Value / time.Duration(intervalFactor)
```

Comment:
Should this not be a multiplication? The intervalFactor is 1,2,3,4,5 or 10. And if understand the frontend code it is factor that "lowers the resultion" by that amount, ie increasing the step by that amount: 

also intervalFactor is not a duration. 

https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/prometheus/datasource.ts#L162

Reply:
Yes! Brain fart....

I cast it to duration to make it possible to divide (soon multiply) it with interval (time.Duration). Casting interval to int64 and then back to time.Duration felt noisy.